cylc/flow/cfgspec/workflow.py

```diff
@@ -996,6 +996,144 @@ def get_script_common_text(this: str, example: Optional[str] = None):
+                   release of Cylc.
+
+                By default, the completion condition ensures that all required
+                outputs, i.e. outputs which appear in the graph but are not
+                marked as optional with the ``?`` character, are completed.
+
+                E.g., in this example, the task ``foo`` must generate the
+                required outputs ``succeeded`` and ``x`` and it may or may not
+                generate the optional output ``y``:
+
+                .. code-block:: cylc-graph
+
+                   foo => bar
+                   foo:x => x
+                   foo:y? => y
+
+                In Python syntax that condition looks like this:
+
+                .. code-block:: python
+
+                   # the task must succeed and generate the custom output "x"
+                   succeeded and x
+
+                The ``completion`` configuration allows you to override the
+                default completion to suit your needs.
+
+                E.g., in this example, the task ``foo`` has three optional
+                outputs, ``x``, ``y`` and ``z``:
+
+                .. code-block:: cylc-graph
+
+                   foo:x? => x
+                   foo:y? => y
+                   foo:z? => z
+                   x | y | z => bar
+
+                Because all three of these outputs are optional, if none of
+                them are generated, the task's outputs will still be
+                considered complete.
+
+                If you wanted to require that at least one of these outputs is
+                generated you can configure the completion condition like so:
+
+                .. code-block:: python
+
+                   # the task must succeed and generate at least one of the
+                   # outputs "x" or "y" or "z":
+                   succeeded and (x or y or z)
+
+                .. note::
+
+                   For the completion expression, hyphens in task outputs
+                   are converted into underscores to allow
+                   evaluation by Python, e.g.:
+
+                   .. code-block:: cylc
+
+                      [runtime]
+                          [[foo]]
+                              completion = succeeded and my_output # underscore
+                              [[[outputs]]]
+                                  my-output = 'my custom task output' # hyphen
+
+                .. note::
+
+                   In some cases the succeeded output might not explicitly
+                   appear in the graph, e.g:
+
+                   .. code-block:: cylc-graph
+
+                      foo:x? => x
+
+                   In these cases succeess is presumed to be required unless
+                   explicitly stated otherwise, either in the graph:
+
+                   .. code-block:: cylc-graph
+
+                      foo?
+                      foo:x? => x
+
+                   Or in the completion expression:
```

Comment:
```suggestion
                   Or by omitting it in the completion expression:
```

Reply:
This suggestion is not correct as `succeed` may be present in the completion providing that it is optional.

Reply:
Ah right. The example given
```
completion = x  # no reference to succeeded here
```
still kind of implies that you must omit it. I'm trying to think if this could be worded differently

Reply:
```suggestion
                   Or in the completion expression (either by omitting it or
                   including it with an `or` statement):
```